### PR TITLE
fix compile errors on NodeJS 12

### LINKED ISSUE #92: NodeJS 12.1.0
Now that NodeJS 12.1.0 has been officially released. Is there any plan to support it?

### PR DESCRIPTION
**Pull request by felixhao28**
Fix #92

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,11 +25,17 @@
     "async"
   ],
   "devDependencies": {
-    "async": "~0.9.0",
-    "jasmine-focused": "~1.0.7"
+    "async": "~2.6.2",
+    "grunt": "^1.0.4",
```
Comment (mofux) [inline package.json]: Why did you add the `grunt` dependencies?

File: package.json [hunk 2]
```diff
@@ -25,11 +25,17 @@
...
+    "grunt-cli": "^1.3.2",
+    "grunt-coffeelint": "0.0.16",
+    "grunt-contrib-coffee": "^2.1.0",
+    "grunt-shell": "^3.0.1",
+    "jasmine-focused": "~1.0.7",
+    "node-cpplint": "^0.4.0"
```
Comment (mofux) [inline package.json]: Same as above, `node-cpplint` does not seem to be a required dependency

File: src/onig-scanner.cc [hunk 3]
```diff
@@ -88,8 +92,9 @@ Local<Value> OnigScanner::FindNextMatchSync(OnigString* source, Local<Number> v8
   shared_ptr<OnigResult> bestResult = searcher->Search(source, charOffset);
   if (bestResult != NULL) {
     Local<Object> result = Nan::New<Object>();
-    result->Set(Nan::New<String>("index").ToLocalChecked(), Nan::New<Number>(bestResult->Index()));
-    result->Set(Nan::New<String>("captureIndices").ToLocalChecked(), CaptureIndicesForMatch(bestResult.get(), source));
+	Local<Context> context = Nan::GetCurrentContext();
```
Comment (mofux) [inline src/onig-scanner.cc]: The indentation should be fixed here

Comment (felixhao28) [reply src/onig-scanner.cc]: good catch

File: package.json (hunk 1 above)
Comment (felixhao28) [reply package.json]: Because the readme says `Run grunt to compile the CoffeeScript and native code` and apparently you need to install grunt and other related dependencies before running grunt. So I added them to devDependencies for future developers. No idea why they were not added earlier.

**Comment by rafeca** [comment]
Thanks for the PR @felixhao28 !

`grunt` is not necessary on this repo (the Readme instructions were obsolete 😅 ).

I've updated your PR to remove the `grunt` dependencies and remove the Node v4 builds on Travis (since they don't pass).

Once the CI builds pass, I'll merge this PR and publish a new version of the package

**Comment by felixhao28** [comment]
@rafeca @alexandrudima when I update this dependency of microsoft/vscode-textmate from 7.0.0 to 7.2.0, one test breaks. I have yet to find out the real cause behind it. This blocks microsoft/vscode-textmate from running in Node 12

image
